### PR TITLE
feat(dataplex): add code samples for Aspect Types

### PR DESCRIPTION
**Pull request by jacspa96**
## Description

Fixes b/368503059

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [X] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [X] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [X] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [X] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -62,6 +62,7 @@
 /pubsub/**/*                           @GoogleCloudPlatform/api-pubsub-and-pubsublite @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
 /pubsublite/**/*                       @GoogleCloudPlatform/api-pubsub-and-pubsublite @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
 /managedkafka/**/*                     @GoogleCloudPlatform/api-pubsub-and-pubsublite @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
+/dataplex/**/*                         @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers @GoogleCloudPlatform/googleapi-dataplex
```
Comment (jacspa96) [inline .github/CODEOWNERS:65]: @m-strzelczyk does [python-docs-samples](https://github.com/GoogleCloudPlatform/python-docs-samples) work differently than [java-docs-samples](https://github.com/GoogleCloudPlatform/java-docs-samples)? In java's CODEOWNERS it was correct to add @GoogleCloudPlatform/googleapi-dataplex as an owner (https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS#L92)

Comment (iennae) [reply .github/CODEOWNERS:65]: the team has to be added to the repo. I'll sort that out.

Comment (iennae) [reply .github/CODEOWNERS:65]: the remaining error was with another line in the file, checked and the account is no longer Google affiliated so have removed.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -38,7 +38,7 @@
 # Platform Ops
 /monitoring/opencensus                 @yuriatgoogle @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
 /monitoring/prometheus                 @yuriatgoogle @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
-/trace/**/*                            @ymotongpoo @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
+/trace/**/*                            @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
```
Comment (m-strzelczyk) [inline .github/CODEOWNERS:41]: would be great to have some dedicated team for this

File: .github/CODEOWNERS (hunk 1 above)
Comment (m-strzelczyk) [inline .github/CODEOWNERS:65]: same as above - we try to avoid folders without narrowed down owners.

File: .github/CODEOWNERS (hunk 2 above)
Comment (m-strzelczyk) [reply .github/CODEOWNERS:41]: Would `googleapi-dataplex`be a good owner?

File: dataplex/snippets/aspect_type_test.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    expected_aspect_type_to_create = f"projects/{PROJECT_ID}/locations/{LOCATION}/aspectTypes/{aspect_type_id_to_create}"
+    aspect_type = create_aspect_type.create_aspect_type(
+        PROJECT_ID, LOCATION, aspect_type_id_to_create, []
+    )
+    # Clean-up created Aspect Type
+    delete_aspect_type.delete_aspect_type(
+        PROJECT_ID, LOCATION, aspect_type_id_to_create
+    )
+
+    assert expected_aspect_type_to_create == aspect_type.name
```
Comment (m-strzelczyk) [inline dataplex/snippets/aspect_type_test.py:79]: We usually do it like:

```
try:
    create()
    assert
finally:
    delete()
```

**Review by m-strzelczyk — CHANGES_REQUESTED** [review]
Very nice code!

Main issue is with the folder ownership in CODEOWNERS - would be best to have a team dedicated to those folders assigned :)

File: .github/CODEOWNERS (hunk 2 above)
Comment (jacspa96) [reply .github/CODEOWNERS:41]: Please check resolved comment in this PR for reference

File: .github/CODEOWNERS (hunk 1 above)
Comment (jacspa96) [reply .github/CODEOWNERS:65]: But here `googleapi-dataplex` is specified as owner. Did you miss it or are you referring to something else?

Comment (m-strzelczyk) [reply .github/CODEOWNERS:65]:
```suggestion
/dataplex/**/*                         @GoogleCloudPlatform/googleapi-dataplex @GoogleCloudPlatform/python-samples-reviewers @GoogleCloudPlatform/cloud-samples-reviewers
```

I missed it, sorry :) Also, the smallest - most specific team should go first, them the more generic ones, as it's in this order automations look for approvers etc.

File: .github/CODEOWNERS (hunk 2 above)
Comment (m-strzelczyk) [reply .github/CODEOWNERS:41]: OK, that's Fine :) sorry, didn't read previous comments

**Review by m-strzelczyk — APPROVED** [review]

**Pull request merged**